### DISCUSSION TITLE
How to disable canvas panning when dragging over a non-draggable node?

### DISCUSSION THREAD

**Discussion by hardyyb2** [xyflow/xyflow#4622] [Q&A] [answered]
Hi team,

I'm currently using `reactflow` in my project and have enabled `panOnDrag` to allow the canvas to pan when dragging. However, I want to refine this behaviour so that the canvas only pans when dragging directly on the canvas, not when dragging a node. The reason for this is I want to implement a custom drag-and-drop behaviour for nodes (so no native dragging), allowing them to be moved within the canvas without triggering the panning.

Here’s what I'm trying to achieve:

* Canvas panning should work when dragging on empty areas of the canvas.
* Canvas should not pan when a node is being dragged.

Is there any native or simple way to achieve this?

Thank you in advance

**Comment by bcakmakoglu** [comment, marked as answer]
You can just add `className: 'nopan'` to the nodes

```ts
  {
    id: "a",
    position: { x: 0, y: 0 },
    data: { label: "Hello World" },
    className: "nopan",
  },
```

[Here's an example.](https://codesandbox.io/p/sandbox/rough-morning-qzzcq8?file=%2Fsrc%2Fnodes%2Findex.ts%3A14%2C1-20%2C5)

**Reply by hardyyb2** [reply]
Awesome, that works. Thank you very much
